lib/cocoapods/generator/xcconfig/pod_xcconfig.rb

```diff
@@ -69,11 +69,13 @@ def generate
             @xcconfig.merge!(file_accessor.spec_consumer.pod_target_xcconfig)
           end
           XCConfigHelper.add_target_specific_settings(target, @xcconfig)
-          @xcconfig.merge! XCConfigHelper.settings_for_dependent_targets(target, target.recursive_dependent_targets)
+          recursive_dependent_targets = target.recursive_dependent_targets
+          @xcconfig.merge! XCConfigHelper.settings_for_dependent_targets(target, recursive_dependent_targets)
           if @test_xcconfig
-            @xcconfig.merge! XCConfigHelper.settings_for_dependent_targets(target, [target, *target.test_dependent_targets])
-            XCConfigHelper.generate_vendored_build_settings(nil, [target, *target.test_dependent_targets], @xcconfig)
-            XCConfigHelper.generate_other_ld_flags(nil, [target, *target.test_dependent_targets], @xcconfig)
+            test_dependent_targets = [target, *target.test_dependent_targets]
+            @xcconfig.merge! XCConfigHelper.settings_for_dependent_targets(target, test_dependent_targets - recursive_dependent_targets)
```

Comment:
this is the real change `test_dependent_targets - recursive_dependent_targets`

`recursive_dependent_targets` have been added above on line 73.